react-native-ssi/ios/RNSsi.mm

```diff
@@ -9,13 +9,17 @@ @implementation RNSsi
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)
 {
-    NSError *error = [[NSError alloc] init];
-    
-    @try  {
-        SsiDIDKeyWrapper *thing = SsiGenerateDIDKey(keyType, &error);
-        resolve(thing.didKey);
-    } @catch (NSException *exception) {
-        reject(@"Something weng wrong", @"wrong", error);
+    NSError *error;
+    NSData *bytes = SsiGenerateDIDKey(keyType, &error);
+    if (error != nil) {
+        return reject(@"RNSsi", @"error generating DID key", error);
+    }
+
+    NSMutableDictionary *json = [NSJSONSerialization JSONObjectWithData:bytes options:NSJSONReadingMutableContainers error:&error];
+    if (error != nil) {
```

Comment:
should this point to a different error object or does that not matter?

Reply:
We could, but don't need to in this situation. Error handling in Obj-C is really weird.

Because we've explicitly checked that `error != nil` after `SsiGenerateDIDKey`, any new error created would have been generated by the `[NSJSONSerialization JSONObjectWithData:...` above.

Reply:
Makes sense